src/realm/query_engine.hpp

```diff
@@ -1305,6 +1305,168 @@ class TimestampNode : public TimestampNodeBase {
     }
 };
 
+class DecimalNodeBase : public ParentNode {
+public:
+    using TConditionValue = Decimal128;
+    static const bool special_null_node = false;
+
+    DecimalNodeBase(Decimal128 v, ColKey column)
+        : m_value(v)
+    {
+        m_condition_column_key = column;
+    }
+
+    DecimalNodeBase(null, ColKey column)
+        : DecimalNodeBase(Decimal128{}, column)
```

Comment:
This should be `Decimal128{null}` the default constructor was changed to be a value of zero.

Reply:
Dammit. I had 'null` here at some point, but it did not compile. Then I removed it to get the rest working, but forgot to come back. The same for ObjectId (copy-paste).